# this patch does two things: 
# 1. replace gcc by g++ for getting the version (otherwise 
#    both compilers would be necessary for compiling)
# 2. gclib does not provide the objects/libraries necessary
#    for building gffcompare, so they are added as shared 
#    object
--- /tmp/Makefile	2018-06-26 00:26:41.000000000 +0200
+++ Makefile	2018-11-29 13:00:38.899628706 +0100
@@ -41,10 +41,10 @@
 CC      := g++
 
 BASEFLAGS  := -Wall -Wextra ${SEARCHDIRS} $(MARCH) \
- -D_REENTRANT -fno-exceptions -fno-rtti
+ -D_REENTRANT -fno-exceptions -fno-rtti -fPIC
 
 #add the link-time optimization flag if gcc version > 4.5
-GCC_VERSION:=$(subst ., ,$(shell gcc -dumpversion))
+GCC_VERSION:=$(subst ., ,$(shell g++ -dumpversion))
 GCC_MAJOR:=$(word 1,$(GCC_VERSION))
 GCC_MINOR:=$(word 2,$(GCC_VERSION))
 #GCC_SUB:=$(word 3,$(GCC_VERSION))
@@ -73,14 +73,17 @@
 LINKER  := g++
 LIBS := 
 OBJS := GBase.o GStr.o GArgs.o
+GCLIBS := GFastaIndex.o GFaSeqGet.o gff.o gdna.o codons.o GBase.o GStr.o GArgs.o
 
 .PHONY : all
-all:    gtest threads
+all:    gtest threads gffcomparelibs
 
 version: ; @echo "GCC Version is: "$(GCC_MAJOR)":"$(GCC_MINOR)":"$(GCC_SUB)
 	@echo "> GCC Opt. string is: "$(GCC45OPTS)
 debug:  gtest threads
 nodebug:  gtest threads
+gffcomparelibs: ${GCLIBS}
+	${LINKER} ${LDFLAGS} $(GCC45OPTS) $(GCC45OPTMAIN) -shared -o libgffc.so ${GCLIBS}
 gtest.o : GBase.h GArgs.h GVec.hh GList.hh GBitVec.h
 GArgs.o : GArgs.h
 gtest: $(OBJS) gtest.o
